backend/src/bundles/lms-data/lms-data.service.ts

```diff
@@ -0,0 +1,135 @@
+import { userRepository } from '~/bundles/users/users.js';
+import { LMSDataApiPath } from '~/common/enums/enums.js';
+import { http } from '~/common/packages/http/http.js';
+import { type Service } from '~/common/types/types.js';
+
+import { LMSDataEntity } from './lms-data.entity.js';
+import { type LMSDataRepository } from './lms-data.repository.js';
+import {
+    type LMSDataServerResponseDto,
+    type UserLMSDataDto,
+} from './types/types.js';
+
+class LMSDataService implements Service {
+    private lmsDataRepository: LMSDataRepository;
+    private requestsToLMSHeaders: { 'X-Token': string | undefined };
+
+    public constructor(
+        lmsDataRepository: LMSDataRepository,
+        token: string | undefined,
+    ) {
+        this.lmsDataRepository = lmsDataRepository;
+        this.requestsToLMSHeaders = { 'X-Token': token };
+    }
+
+    public async findByUserId(
+        userId: string,
+    ): Promise<UserLMSDataDto | undefined> {
```

Comment:
Why this return undefined?
You can make throw HttError(...)

Reply:
will fix it